load_tests/load_test.py

```diff
@@ -53,17 +68,38 @@ def calculate_total_input_number(throughput):
 
 # 1. Configure task definition for each load test based on existing templates
 # 2. Register generated task definition
-def generate_task_definition(throughput):
+def generate_task_definition(throughput, input_logger, s3_fluent_config_arn):
+    if not hasattr(generate_task_definition, "counter"):
+        generate_task_definition.counter = 0  # it doesn't exist yet, so initialize it
+    generate_task_definition.counter += 1
+    destination_identifier = get_destination_identifier(throughput, input_logger)
+    destination_identifier_firelens = get_destination_identifier_firelens(throughput, input_logger)
     task_definition_dict = {
+
+        # App Container Environment Variables
+        '$APP_IMAGE': input_logger['logger_image'],
+        '$LOGGER_RUN_TIME_IN_SECOND': str(LOGGER_RUN_TIME_IN_SECOND),
+        
+        # Firelens Container Environment Variables
+        '$FLUENT_BIT_IMAGE': os.environ['FLUENT_BIT_IMAGE'],
+        '$INPUT_NAME': input_logger['name'],
+        '$LOGGER_PORT': "4560",
+        '$FLUENT_CONFIG_S3_FILE_ARN': s3_fluent_config_arn,
+        '$OUTPUT_PLUGIN': OUTPUT_PLUGIN,
+
+        # General Environment Variables
+        '$FIRELENS_DESTINATION_IDENTIFIER': destination_identifier_firelens,
+        '$DESTINATION_IDENTIFIER': destination_identifier,
         '$THROUGHPUT': throughput,
+
+        # Task Environment Variables
         '$TASK_ROLE_ARN': os.environ['LOAD_TEST_TASK_ROLE_ARN'],
         '$TASK_EXECUTION_ROLE_ARN': os.environ['LOAD_TEST_TASK_EXECUTION_ROLE_ARN'],
-        '$FLUENT_BIT_IMAGE': os.environ['FLUENT_BIT_IMAGE'],
-        '$APP_IMAGE': os.environ['APP_IMAGE'],
-        '$LOGGER_RUN_TIME_IN_SECOND': str(LOGGER_RUN_TIME_IN_SECOND),
+
+        # Plugin Specific Environment Variables
         'cloudwatch': {'$CW_LOG_GROUP_NAME': os.environ['CW_LOG_GROUP_NAME']},
-        'firehose': {'$DELIVERY_STREAM': os.environ[f'FIREHOSE_TEST_{throughput}']},
-        'kinesis': {'$STREAM': os.environ[f'KINESIS_TEST_{throughput}']},
+        'firehose': {'$DELIVERY_STREAM_PREFIX': f'{PREFIX}{PLATFORM}-firehoseTest-deliveryStream'},
```

Comment:
we can delete this part if you set the env vars like this: https://github.com/aws/aws-for-fluent-bit/blob/mainline/load_tests/setup_test_environment.sh#L12

Reply:
I changed it from stream to stream_prefix, since there are now the following streams

```
# Kinesis
load-test-fluent-bit-ecs-kinesisStream-20m
load-test-fluent-bit-ecs-kinesisStream-25m
load-test-fluent-bit-ecs-kinesisStream-30m

load-test-fluent-bit-ecs-kinesisStream-std-20m
load-test-fluent-bit-ecs-kinesisStream-std-25m
load-test-fluent-bit-ecs-kinesisStream-std-30m

# Firehose
load-test-fluent-bit-ecs-firehoseTest-deliveryStream-20m
load-test-fluent-bit-ecs-firehoseTest-deliveryStream-25m
load-test-fluent-bit-ecs-firehoseTest-deliveryStream-30m

load-test-fluent-bit-ecs-kinesisTest-deliveryStream-std-20m
load-test-fluent-bit-ecs-kinesisTest-deliveryStream-std-25m
```

If we add eks, then we'll need to multiply these streams yet again. to: 
```
# Kinesis
load-test-fluent-bit-eks-kinesisStream-20m
[...]
```

I would like the env vars set on this script to be as minimal as possible. Since we can fully derive the stream prefix from Prefix and Platform, I think that we shouldn't pollute the env vars with combined information, as to make things more complicated. Otherwise it's going to become difficult when we have multiple platforms, prefixes, and postfixes ("" and "-std-")

What do you think?

Reply:
I agree. So we can delete the duplicate part in `setup_test_environment.sh`?

Reply:
Keeping it simple for now and deleting it from setup_test_environment.sh. As discussed off line, it may be a good idea to have a resource name provider python module that is shared between the creation and load test script. This module could contain 
1. parameter values: throughput, platform, etc
2. parameter->name functions

That way we can abstract the parameter values, such as platforms, from the resources.

Will hold off from this change till a load test improvement PR.

Reply:
Added to load test improvement PR